File: src/main/java/com/vaadin/demo/component/rangeslider/RangeSliderValueAlwaysVisible.java
package com.vaadin.demo.component.rangeslider;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.slider.RangeSlider;
import com.vaadin.flow.component.slider.RangeSliderValue;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("range-slider-value-always-visible")
public class RangeSliderValueAlwaysVisible extends Div {

    public RangeSliderValueAlwaysVisible() {
        // tag::snippet[]
        RangeSlider rangeSlider = new RangeSlider("Brightness",
                0, 100, new RangeSliderValue(25, 75));
        // rangeSlider.setValueAlwaysVisible(true);
        add(rangeSlider);
        // end::snippet[]
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<RangeSliderValueAlwaysVisible> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/range-slider/range-slider-value-always-visible.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/slider/vaadin-range-slider.js';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('range-slider-value-always-visible')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-range-slider
        label="Brightness"
        min="0"
        max="100"
        .value="${[25, 75]}"
        value-always-visible
      ></vaadin-range-slider>
      <!-- end::snippet[] -->
    `;
  }
}
